### DISCUSSION TITLE
Shell Navigation with Models

### DISCUSSION THREAD

**Discussion by ziomek64** [dotnet/maui#14045] [Ideas] [answered]
Was this ever added? I remember long time ago there was a plan to add this in Xamarin.Forms but was moved to MAUI? - https://github.com/xamarin/Xamarin.Forms/issues/6848

Is this a thing now or still not? That was my major no-go with shell. Other than that it was really simple to learn in small apps. Can't find anything btw.

**Comment by AndreasReitberger** [comment, marked as answer]
Yes, this is possible. You can pass a `Dictionary<string, object>` parameter now.

```cs
object para1 = new();
Dictionary<string, object> parameters = new() { { "para1", para1 } };

await Shell.Current.GoToAsync("YOUR_URI", parameters: parameters, animate: true);
```

In the `ViewModel`, you can access the parameter like:

```cs
[QueryProperty(nameof(Parameter), "para1")]
    public partial class AppViewModel : BaseViewModel
    {
        #region QueryParameter
        [ObservableProperty]
        protected object parameter;
         //....
     }
```
The parameter can be any class type, for this example it`s just an object.

For docs, see here: https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/shell/navigation?view=net-maui-7.0#pass-data

**Reply by ziomek64** [reply]
So finally any type of custom made object will work? That's the only issue I had working with out of the box xamarin. That's amazing

**Reply by AndreasReitberger** [reply]
Yes, you can pass any custom object with the example above ;)

**Reply by ziomek64** [reply]
Great, can't wait to implement it
